client/src/legacy/CMSMain.js

```diff
@@ -24,20 +24,49 @@ $.entwine('ss', function ($) {
 		}
 	});
 
+  // Page edit mode
+  $('.cms-edit-form').entwine({
+      onadd: function() {
+        // Always use treeview when in page edit mode
+        localStorage.setItem('pages-view-type', 'treeview');
+      }
+  });
+
+  /**
+   * Override super's `onadd` to modify url base the previously select view
+   * type.
+   *
+   * See $('.cms .cms-panel-link') in LeftAndMain.js
+   */
+  $('.cms-panel-deferred.cms-content-view').entwine({
+    onadd: function() {
+      const viewType = localStorage.getItem('pages-view-type');
```

Comment:
I think it'd be a good item to have a fallback, in case there isn't an entry in localStorage